scripts/tests.e2e.sh

```diff
@@ -20,64 +20,70 @@ fi
-curl -L ${DOWNLOAD_URL} -o ${DOWNLOAD_PATH}
-
-echo "extracting downloaded avalanchego"
-if [[ ${GOOS} == "linux" ]]; then
-  tar xzvf ${DOWNLOAD_PATH} -C /tmp
-elif [[ ${GOOS} == "darwin" ]]; then
-  unzip ${DOWNLOAD_PATH} -d /tmp/avalanchego-build
-  mv /tmp/avalanchego-build/build /tmp/avalanchego-v${VERSION}
+AVAGO_FILEPATH=${BASEDIR}/avalanchego-${VERSION}
+if [[ ! -d ${AVAGO_FILEPATH} ]]; then
+  if [[ ! -f ${AVAGO_DOWNLOAD_PATH} ]]; then
+    echo "downloading avalanchego ${VERSION} at ${AVAGO_DOWNLOAD_URL} to ${AVAGO_DOWNLOAD_PATH}"
+    curl -L ${AVAGO_DOWNLOAD_URL} -o ${AVAGO_DOWNLOAD_PATH}
+  fi
+  echo "extracting downloaded avalanchego to ${AVAGO_FILEPATH}"
+  if [[ ${GOOS} == "linux" ]]; then
+    mkdir -p ${AVAGO_FILEPATH} && tar xzvf ${AVAGO_DOWNLOAD_PATH} --directory ${AVAGO_FILEPATH} --strip-components 1
+  elif [[ ${GOOS} == "darwin" ]]; then
+    unzip ${AVAGO_DOWNLOAD_PATH} -d ${AVAGO_FILEPATH}
+    mv ${AVAGO_FILEPATH}/build/* ${AVAGO_FILEPATH}
+    rm -rf ${AVAGO_FILEPATH}/build/
+  fi
+  find ${BASEDIR}/avalanchego-${VERSION}
 fi
-find /tmp/avalanchego-v${VERSION}
+
+AVALANCHEGO_PATH=${AVAGO_FILEPATH}/avalanchego
+AVALANCHEGO_PLUGIN_DIR=${AVAGO_FILEPATH}/plugins
 
 #################################
 # download avalanche-network-runner
 # https://github.com/ava-labs/avalanche-network-runner
-NETWORK_RUNNER_VERSION=1.0.14
-DOWNLOAD_PATH=/tmp/avalanche-network-runner.tar.gz
-DOWNLOAD_URL=https://github.com/ava-labs/avalanche-network-runner/releases/download/v${NETWORK_RUNNER_VERSION}/avalanche-network-runner_${NETWORK_RUNNER_VERSION}_linux_amd64.tar.gz
-if [[ ${GOOS} == "darwin" ]]; then
-  DOWNLOAD_URL=https://github.com/ava-labs/avalanche-network-runner/releases/download/v${NETWORK_RUNNER_VERSION}/avalanche-network-runner_${NETWORK_RUNNER_VERSION}_darwin_amd64.tar.gz
-fi
-
-rm -f /tmp/avalanche-network-runner
-rm -f ${DOWNLOAD_PATH}
-
-echo "downloading avalanche-network-runner ${NETWORK_RUNNER_VERSION} at ${DOWNLOAD_URL}"
-curl -L ${DOWNLOAD_URL} -o ${DOWNLOAD_PATH}
-
-echo "extracting downloaded avalanchego"
-tar xzvf ${DOWNLOAD_PATH} -C /tmp
-/tmp/avalanche-network-runner -h
+ANR_REPO_PATH=github.com/ava-labs/avalanche-network-runner
+ANR_VERSION=v1.2.3
+# version set
+go install -v ${ANR_REPO_PATH}@${ANR_VERSION}
 
 #################################
 echo "building e2e.test"
 # to install the ginkgo binary (required for test build and run)
 go install -v github.com/onsi/ginkgo/v2/ginkgo@v2.0.0
+# Set the CGO flags to use the portable version of BLST                                                                 
+#                                                                                                                       
+# We use "export" here instead of just setting a bash variable because we need                                          
+# to pass this flag to all child processes spawned by the shell.                                                        
+export CGO_CFLAGS="-O -D__BLST_PORTABLE__"
```

Comment:
Could we also add this to `./scripts/build.sh`

Reply:
should also put a new line before and after this chunk

Reply:
there is not `build.sh` script but can create a little one